swagger/src/main/scala/org/http4s/rho/swagger/SwaggerModelsBuilder.scala

```diff
@@ -346,7 +346,8 @@ private[swagger] class SwaggerModelsBuilder(formats: SwaggerFormats) {
   def mkQueryParam[F[_]](rule: QueryMetaData[F, _]): Parameter = {
     val required = !(rule.m.tpe.isOption || rule.default.isDefined)
 
-    TypeBuilder.DataType(rule.m.tpe) match {
+    val tpe = if(rule.m.tpe.isOption) rule.m.tpe.typeArgs.head else rule.m.tpe
```

Comment:
Isn't `typeArgs.head` unsafe is type is an alias like `type OptInt = Option[Int]`? `OptInt` has zero type args.

Reply:
I'll have to check.

Reply:
Maybe `dealias` will be useful here https://github.com/xeno-by/scala/blob/0f4e95574081bd9a945fb5b32d157a32af840cd3/src/reflect/scala/reflect/api/Types.scala#L244

Reply:
You are correct.  I'll write up a test case and do something else.  Good catch @chuwy.

Reply:
So, 
```scala
      type OpSeq = Option[Seq[String]]
      val ra = fooPath +? param[OpSeq]("name") |>> { (s: OpSeq) => "" }
```

Doesn't compile anyways:
```
[error] /home/zarthross/Projects/rho/swagger/src/test/scala/org/http4s/rho/swagger/SwaggerModelsBuilderSpec.scala:105:39: No TypeTag available for OpSeq
[error]       val ra = fooPath +? param[OpSeq]("name") |>> { (s: OpSeq) => "" }
```

Reply:
Hm. I managed to compile it and get `NoSuchElementException`, but admit it was tricky

```
object OtherModule {
  type OpSeq = Option[Int]
  implicit val tt: TypeTag[OpSeq]
}

object OurSpec {
  import OtherModule._

  // remaining specification; fails because of head
}
```

Reply:
And `dealias` helped indeed. I also noticed few other `tpe.head` in that module. Probably we want to fix them as well.

Reply:
well, this is interesting:
```
scala> object Test {
     | type OpSeq = Option[Seq[String]]
     | val t = typeTag[OpSeq]
     | }
defined object Test
```
vs 
```
scala> def test {
     | type OpSeq = Option[Seq[String]]
     | return typeTag[OpSeq]
     | }
<console>:17: error: No TypeTag available for OpSeq
       return typeTag[OpSeq]
```

Reply:
Alright, added a few additional tests around this, and added another dealias that should help

Reply:
I assume that inside `def` `type` is not really global and that somehow prevents compiler from inspecting it. Quite superficial assumption, though, but yeah, that's interesting.